cmd/embedded-cluster/restore.go

```diff
@@ -886,370 +882,399 @@ func installAndWaitForRestoredK0sNode(c *cli.Context, applier *addons.Applier) (
+			os.Setenv("KUBECONFIG", provider.PathToKubeConfig())
+			os.Setenv("TMPDIR", provider.EmbeddedClusterTmpSubDir())
+
+			proxy := getProxySpecFromFlags(c)
+			setProxyEnv(proxy)
+
+			logrus.Debugf("getting restore state")
+			state := getECRestoreState(c.Context)
+			logrus.Debugf("restore state is: %q", state)
+
+			if state != ecRestoreStateNew {
+				shouldResume := prompts.New().Confirm("A previous restore operation was detected. Would you like to resume?", true)
+				logrus.Info("")
+				if !shouldResume {
+					state = ecRestoreStateNew
+				}
 			}
-			if backupToRestore != nil {
-				completionTimestamp := backupToRestore.Status.CompletionTimestamp.Time.Format("2006-01-02 15:04:05 UTC")
-				logrus.Infof("Resuming restore from backup %q (%s)\n", backupToRestore.Name, completionTimestamp)
+			if c.String("airgap-bundle") != "" {
+				logrus.Debugf("checking airgap bundle matches binary")
+				if err := checkAirgapMatches(c); err != nil {
+					return err // we want the user to see the error message without a prefix
+				}
 			}
-		}
 
-		applier, err := getAddonsApplier(c, "", proxy)
-		if err != nil {
-			return err
-		}
-
-		switch state {
-		case ecRestoreStateNew:
-			logrus.Debugf("checking if %s is already installed", binName)
-			if installed, err := isAlreadyInstalled(); err != nil {
-				return err
-			} else if installed {
-				logrus.Errorf("An installation has been detected on this machine.")
-				logrus.Infof("If you want to restore you need to remove the existing installation")
-				logrus.Infof("first. You can do this by running the following command:")
-				logrus.Infof("\n  sudo ./%s reset\n", binName)
-				return ErrNothingElseToAdd
+			// if the user wants to resume, check if a backup has already been picked.
+			var backupToRestore *velerov1.Backup
+			if state != ecRestoreStateNew {
+				logrus.Debugf("getting backup from restore state")
+				var err error
+				backupToRestore, err = getBackupFromRestoreState(c.Context, provider, c.String("airgap-bundle") != "")
+				if err != nil {
+					return fmt.Errorf("unable to resume: %w", err)
+				}
+				if backupToRestore != nil {
+					completionTimestamp := backupToRestore.Status.CompletionTimestamp.Time.Format("2006-01-02 15:04:05 UTC")
+					logrus.Infof("Resuming restore from backup %q (%s)\n", backupToRestore.Name, completionTimestamp)
+
+					runtimeConfig, err = overrideRuntimeConfigFromBackup(c, runtimeConfig, backupToRestore)
+					if err != nil {
+						return fmt.Errorf("unable to override runtime config from backup: %w", err)
+					}
+				}
 			}
 
-			logrus.Infof("You'll be guided through the process of restoring %s from a backup.\n", binName)
-			logrus.Info("Enter information to configure access to your backup storage location.\n")
-			s3Store := newS3BackupStore()
+			// If the installation is available, we can further augment the runtime config from the
+			// installation.
+			if state != ecRestoreStateNew &&
```

Comment:
should this just try to read the installation without checking the state and not fail if it couldn't since it's not critical?

Reply:
feels brittle to have to check states like this as they could change